### PR TITLE
More of GameEngine::Init

### PR DESCRIPTION
**Pull request by jonwil**

### CODE REVIEW

File: src/game/common/recorder.h [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+private:
+    CRCInfo *m_crcInfo;
+    FILE *m_file;
+    Utf8String m_fileName;
+    int m_currentFilePosition;
+    RecorderModeType m_mode;
+    Utf8String m_unk;
+    ReplayGameInfo m_gameInfo;
+    bool m_wasDesync;
+    bool m_doLogging;
```
Comment (tomsons26) [inline src/game/common/recorder.h]: i investigated this a while back, doesn't have anything to do with logging, not sure what its for, class clears map if not set

File: src/game/common/system/radar.h [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    virtual ~Radar() override;
+    virtual void Init() override;
+    virtual void Reset() override;
+    virtual void Update() override;
+
+private:
+    struct RadarEvent
+    {
+        int type;
+        char active;
```
Comment (tomsons26) [inline src/game/common/system/radar.h]: would be bool

File: src/game/common/system/radar.h [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        int create_frame;
+        int die_frame;
+        int fade_frame;
+        RGBAColorInt color1;
+        RGBAColorInt color2;
+        Coord3D world_loc;
+        ICoord2D radar_loc;
+        char sound_played;
```
Comment (tomsons26) [inline src/game/common/system/radar.h]: would be bool

File: src/game/common/system/radar.h [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+    bool m_radarHidden;
+    bool m_radarForceOn;
+    RadarObject *m_objectList;
+    RadarObject *m_localObjectList;
+    float m_terrainAverageZ;
+    float m_waterAverageZ;
+    float m_xSample;
+    float m_ySample;
+    RadarEvent m_event[64];
```
Comment (tomsons26) [inline src/game/common/system/radar.h]: MAX_RADAR_EVENTS

File: src/game/network/gameinfo.h [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#pragma once
+#include "always.h"
+#include "asciistring.h"
+#include "money.h"
+#include "unicodestring.h"
+
+enum GameState
```
Comment (tomsons26) [inline src/game/network/gameinfo.h]: GameState is a class name

File: src/game/network/gameinfo.h [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    virtual void Start_Game(int game_id);
+    virtual bool Am_I_Host() const;
+    virtual int Get_Local_Slot_Num() const;
+    virtual void Reset_Accepted();
+    virtual void Reset_Start_Spots();
+    virtual void Adjust_Slots_For_Map(int i);
+    virtual void Close_Open_Slots();
+
+private:
+    GameState m_state;
```
Comment (tomsons26) [inline src/game/network/gameinfo.h]: this is wrong, set by markPlayerAsPreorder

File: src/game/client/messagestream/metaevent.h [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+#pragma once
+#include "always.h"
+#include "gamemessage.h"
+#include "mempoolobj.h"
+#include "subsysteminterface.h"
+#include "unicodestring.h"
+
+enum MappableKeyCategories
+{
+    CATEGORY_CONTROL = 0x0,
```
Comment (tomsons26) [inline src/game/client/messagestream/metaevent.h]: remove the numbers

File: src/game/client/messagestream/metaevent.h [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+    MK_PGUP = 0xc9,
+    MK_PGDN = 0xd1,
+    MK_INS = 0xd2,
+    MK_DEL = 0xd3,
+    MK_NONE = 0x0,
+};
+
+enum MappableKeyTransition
+{
+    DOWN = 0x0,
```
Comment (tomsons26) [inline src/game/client/messagestream/metaevent.h]: remove the numbers

File: src/game/client/messagestream/metaevent.h [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+    UP = 0x1,
+    DOUBLEDOWN = 0x2,
+};
+
+enum MappableKeyModState
+{
```
Comment (tomsons26) [inline src/game/client/messagestream/metaevent.h:122]:
```
{
    NONE = 0,
    CTRL = 1 << 2,
    ALT = 1 << 6,
    SHIFT = 1 << 4,
    
    CTRL_ALT = CTRL | ALT,
    
    SHIFT_CTRL = SHIFT | CTRL,
    SHIFT_ALT = SHIFT | ALT,
    SHIFT_ALT_CTRL = SHIFT | ALT | CTRL,
};
```

File: src/game/client/messagestream/metaevent.h [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+    SHIFT = 0x10,
+    CTRL_ALT = 0x44,
+    SHIFT_CTRL = 0x14,
+    SHIFT_ALT = 0x50,
+    SHIFT_ALT_CTRL = 0x54,
+};
+
+enum CommandUsableInType
+{
+    COMMANDUSABLE_NONE = 0x0,
```
Comment (tomsons26) [inline src/game/client/messagestream/metaevent.h]: remove the numbers

File: src/game/common/recorder.h (hunk 1 above)
Comment (jonwil) [reply src/game/common/recorder.h]: Renamed it.

File: src/game/common/system/radar.h (hunk 2 above)
Comment (jonwil) [reply src/game/common/system/radar.h]: fixed.

File: src/game/common/system/radar.h (hunk 3 above)
Comment (jonwil) [reply src/game/common/system/radar.h]: fixed.

File: src/game/common/system/radar.h (hunk 4 above)
Comment (jonwil) [reply src/game/common/system/radar.h]: fixed.

File: src/game/network/gameinfo.h (hunk 5 above)
Comment (jonwil) [reply src/game/network/gameinfo.h]: Fixed.

File: src/game/client/messagestream/metaevent.h (hunk 7 above)
Comment (jonwil) [reply src/game/client/messagestream/metaevent.h]: fixed.

File: src/game/network/gameinfo.h (hunk 6 above)
Comment (jonwil) [reply src/game/network/gameinfo.h:78]: fixed.

File: src/game/client/messagestream/metaevent.h (hunk 8 above)
Comment (jonwil) [reply src/game/client/messagestream/metaevent.h]: fixed.

File: src/game/client/messagestream/metaevent.h (hunk 9 above)
Comment (jonwil) [reply src/game/client/messagestream/metaevent.h:122]: fixed.

File: src/game/client/messagestream/metaevent.h (hunk 10 above)
Comment (jonwil) [reply src/game/client/messagestream/metaevent.h]: fixed.

File: src/game/network/gameinfo.h (hunk 6 above)
Comment (jonwil) [reply src/game/network/gameinfo.h]: fixed.

File: src/game/client/messagestream/metaevent.h [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+    COMMANDUSABLE_SHELL,
+    COMMANDUSABLE_GAME,
+};
+
+class MetaMapRec : public MemoryPoolObject
+{
+    IMPLEMENT_POOL(MetaMapRec);
+
+public:
+    virtual ~MetaMapRec() override;
```
Comment (xezon) [inline src/game/client/messagestream/metaevent.h:147]: Constructor will come later?

File: src/game/client/messagestream/metaevent.h [hunk 12]
```diff
@@ -0,0 +1,177 @@
...
+
+    MetaMapRec *m_next;
```
Comment (xezon) [inline src/game/client/messagestream/metaevent.h:149]: Members are not private?

File: src/game/common/recorder.h [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+class ReplayGameInfo : public GameInfo
+{
+private:
+    GameSlot m_ReplaySlot[8];
+};
+
+class RecorderClass : public SubsystemInterface
+{
+public:
+    virtual ~RecorderClass() override;
```
Comment (xezon) [inline src/game/common/recorder.h:37]: Constructor will come later?

File: src/game/common/system/radar.h [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+    virtual void Xfer_Snapshot(Xfer *xfer) override;
+    virtual void Load_Post_Process() override;
+    virtual void Refresh_Terrain(TerrainLogic *logic);
+    virtual void Queue_Terrain_Refresh();
+    virtual void New_Map(TerrainLogic *logic);
+    virtual void Draw(int i1, int i2, int i3, int i4) = 0;
+    virtual void Clear_Shroud() = 0;
+    virtual void Set_Shroud_Level(int i1, int i2, CellShroudStatus status) = 0;
+
+    virtual ~Radar() override;
```
Comment (xezon) [inline src/game/common/system/radar.h:40]: Is there a reason destructor is ordered behind? Typically it is put at the top of virtuals list.

File: src/game/common/system/radar.h [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+
+class Radar : public SnapShot, public SubsystemInterface
+{
+public:
+    virtual void CRC_Snapshot(Xfer *xfer) override;
+    virtual void Xfer_Snapshot(Xfer *xfer) override;
+    virtual void Load_Post_Process() override;
+    virtual void Refresh_Terrain(TerrainLogic *logic);
+    virtual void Queue_Terrain_Refresh();
+    virtual void New_Map(TerrainLogic *logic);
```
Comment (xezon) [inline src/game/common/system/radar.h:35]: 3x not pure virtual?

File: src/game/logic/scriptengine/victoryconditions.h [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+#pragma once
+#include "always.h"
+#include "subsysteminterface.h"
+
+class Player;
+
+class VictoryConditionsInterface : public SubsystemInterface
+{
+public:
+    virtual ~VictoryConditionsInterface() override;
```
Comment (xezon) [inline src/game/logic/scriptengine/victoryconditions.h:24]: Constructor missing?

File: src/game/client/messagestream/metaevent.h (hunk 11 above)
Comment (jonwil) [reply src/game/client/messagestream/metaevent.h:147]: Yes.

File: src/game/common/recorder.h (hunk 13 above)
Comment (jonwil) [reply src/game/common/recorder.h:37]: Yes

File: src/game/common/system/radar.h (hunk 15 above)
Comment (jonwil) [reply src/game/common/system/radar.h:35]: These are not pure virtual.

File: src/game/logic/scriptengine/victoryconditions.h (hunk 16 above)
Comment (jonwil) [reply src/game/logic/scriptengine/victoryconditions.h:24]: Constructor will come later.

File: src/game/client/messagestream/metaevent.h (hunk 12 above)
Comment (jonwil) [reply src/game/client/messagestream/metaevent.h:149]: They are not as far as I can see.

File: src/game/common/system/radar.h (hunk 14 above)
Comment (jonwil) [reply src/game/common/system/radar.h:40]: Destructor is in the second vtable so it makes sense to order it this way.

**Review by xezon — APPROVED** [review]
Logically looks proper.

**Pull request merged**